### PR TITLE
Integration test using dummy VDAF with agg param

### PR DESCRIPTION
**Pull request by tgeoghegan**
Enable Janus to run the dummy VDAF from crate `prio` with arbitrary rounds
(by extending the representation of `VdafInstance::Fake`) and with multiple
aggregation parameters chosen by the test.

To that end, this resurrects some functionality previously deleted in #1508.
At the time, we had that code gated on `#[cfg(test)]`, with some cautions
about its likely performance problems. I'm bringing it back in the release
configuration of Janus. I don't think we need a feature flag for this: the
various `VdafInstance::Fake` variants that would use these codepaths are not
currently compiled into release Janus builds. Even if they were, deployments
can avoid this risky code by simply not using those VDAFs.

This wires up some tests for both time interval and fixed size tasks, running
multiple collections against a single batch with varying aggregation
parameters. However, a few tests are disabled because of stuff that doesn't
work yet:

- creating fixed size aggregation jobs w/ agg param
- using `Vdaf.is_valid` to check whether subsequent collections against a
  single batch are valid

Part of #225

- [x] Release a `prio` with https://github.com/divviup/libprio-rs/pull/970 and depend on that in this PR
- [x] Rebase after we take #2782

### CODE REVIEW

File: aggregator/src/aggregator/aggregation_job_creator.rs
Comment (branlwyd) [inline aggregator/src/aggregator/aggregation_job_creator.rs:3420]: I think this comment goes above the block above which checks if the report is scrubbed.

File: aggregator/src/aggregator/aggregation_job_creator.rs
Comment (branlwyd) [inline aggregator/src/aggregator/aggregation_job_creator.rs:3415]: nit: a cleaner way to implement this would be with a trait (probably an extension of `VdafHasAggregationParameter`), but I think it's OK for test code.

but maybe an easier-to-follow check would be: `TypeId::of::<A::AggregationParam>() == TypeId::of::<()>()`? (https://doc.rust-lang.org/std/any/struct.TypeId.html) That would make it clearer this is a function of the VDAF in use, rather than something that might change per-iteration.

File: aggregator_core/src/datastore.rs
Comment (branlwyd) [inline aggregator_core/src/datastore.rs:1258]: Q: is there a functional need to set these as `aggregation_started`? or is this bookkeeping?

File: messages/Cargo.toml
Comment (branlwyd) [inline messages/Cargo.toml:23]: Note to self: needs to be resolved before this can merge.

File: Cargo.toml
Comment (tgeoghegan) [reply Cargo.toml:50]: Yup, I put a task list in the PR description.

File: aggregator/src/aggregator/aggregation_job_creator.rs
Comment (tgeoghegan) [reply aggregator/src/aggregator/aggregation_job_creator.rs:3408]: You are correct that we don't need the type annotation on the `HashSet<_>` on L3403. However, just as on L3383, I need to cast to `&A::AggregationParam` or the compiler won't let me call `clone`. I don't really understand why, TBH.

File: aggregator_core/src/datastore.rs
Comment (tgeoghegan) [reply aggregator_core/src/datastore.rs]: No, this just got resurrected from #1508, but it doesn't do anything anymore. I simplified the query considerably.

File: aggregator_core/src/datastore.rs
Comment (tgeoghegan) [reply aggregator_core/src/datastore.rs]: Good point. I left a comment suggesting that.

File: aggregator_core/src/datastore.rs
Comment (tgeoghegan) [reply aggregator_core/src/datastore.rs:1259]: I spoke too soon: the integration test fails if we don't set `aggregation-started = TRUE`. I believe I'll need to revisit this as I continue in this area (because in the presence of varying aggregation parameters, a single boolean doesn't suffice) but for now I'll leave the query as-is.

File: aggregator_core/src/datastore/models.rs
Comment (branlwyd) [inline aggregator_core/src/datastore/models.rs:365]: I think we made this change because the aggregation parameter was potentially sensitive (e.g., IIUC, for Poplar1/Mastic it would store prefixes of measurement values); I think we want to avoid logging aggregation parameter values.

If we really need to log these, maybe we could:
a) log some safe value derived from the aggregation parameter, e.g. a hash of the value
b) require that the `Debug` implementations of aggregation parameters take responsibility for producing non-sensitive debug values in a VDAF-specific way

(b) is the best IMO but also the hardest to implement, and we'd need to be careful not to backslide as new VDAF implementations come around.

File: aggregator/src/aggregator/aggregation_job_creator.rs
Comment (branlwyd) [reply aggregator/src/aggregator/aggregation_job_creator.rs:3548]: I can reproduce this. Very strange, but minor enough not to be worth spending too much time on.

File: aggregator/src/aggregator/aggregation_job_creator.rs
Comment (branlwyd) [inline aggregator/src/aggregator/aggregation_job_creator.rs:3560]: This comment doesn't match the code logic. The current code logic does nothing on each iteration unless the aggregation param is the unit type.

File: aggregator/src/aggregator/aggregation_job_creator.rs
Comment (branlwyd) [inline aggregator/src/aggregator/aggregation_job_creator.rs:3555]: nit: hoist these aggregation-param-is-unit-type checks outside of the loops they are contained in, since these are effectively constant values.

File: aggregator_core/src/datastore/models.rs
Comment (tgeoghegan) [reply aggregator_core/src/datastore/models.rs:365]: I did this because I wanted to see what the aggregation param value being used with the dummy VDAF is, and that type is simple enough that including it in Debug output wasn't too noisy. I agree that (b) is desirable, but currently `Poplar1AggregationParam`'s debug output contains too much noise. So I'm going to revert this.

File: aggregator/src/aggregator/aggregation_job_creator.rs
Comment (tgeoghegan) [reply aggregator/src/aggregator/aggregation_job_creator.rs:3548]: If you have a small reproduction case, we should file a bug on `rustc`, because this doesn't make a lot of sense to me.

**Review by branlwyd — APPROVED** [review]
LGTM once we have a released prio.

**Pull request merged**
